tools/automation/src/pr-check-icons-extensions-404.ts

```diff
@@ -0,0 +1,94 @@
+/********************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import * as fs from 'fs-extra';
+import * as glob from 'glob';
+import * as jsyaml from 'js-yaml';
+
+import Axios from 'axios';
+
+export interface Extension {
+  name?: string;
+  version?: string;
+  icon?: string;
+  vsix?: string[];
+  error: boolean;
+  errorMessage?: string;
+}
+
+const metaYamlFiles = glob.sync('./../../v3/plugins/**/*.yaml');
+
+async function checkAllPlugins() {
+  const extensions: Extension[] = await Promise.all(
+    metaYamlFiles.map(
+      async (metaYamlFile): Promise<Extension> => {
+        let metaYaml = await fs.readFile(metaYamlFile, 'utf-8');
+        let metaYamlString;
+        let extension: Extension = {
+          error: false,
+        };
+        try {
+          metaYamlString = jsyaml.safeLoad(metaYaml, { schema: jsyaml.JSON_SCHEMA });
+        } catch (yamlErr) {
+          extension.error = true;
+          extension.errorMessage = `Error parsing YAML for ${metaYamlFile}: ${yamlErr}`;
+          return extension;
+        }
+
+        if (metaYamlString) {
+          let metaYamlObject: { icon?: string; name: string; version: string; spec?: { extensions?: [string] } };
+          metaYamlObject = JSON.parse(JSON.stringify(metaYamlString, null, 2));
```

Comment:
curious: why do we stringify the json string ? and why also we need to pretty format it ?

Reply:
AFAIU, js-yaml returns any of the following [string | object | undefined], so it gets converted to a JSON string then an object we can use.

I can remove the pretty formatting though that was unneeded.